File: bin/LeetCode/Arrays/Medium/ContainsDuplicate.js
const nums = [1, 1, 1, 3, 3, 4, 3, 2, 4, 2];
const containsDuplicate = (nums) => {
  let ans = 0;
  for (let i = 0; i < nums.length; i++) {
    ans = ans ^ nums[i];
  }
  for (let i = 1; i < nums.length; i++) {
    ans = ans ^ i;
  }
  return ans;
};
console.log(containsDuplicate(nums));

File: Algos/SlidingWindow/ContainsDuplicate.java
package Algos.SlidingWindow;

import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {
    public static boolean containsNearbyDuplicate(int[] nums, int k) {
        Set<Integer> set = new HashSet<>();

        for (int j = 0; j < nums.length; j++) {
            if (j > k) {
                set.remove(nums[j - k - 1]);
            }

            if (!set.add(nums[j])) {
                return true;
            }
        }
        return false;
    }

    public static void main(String[] args) {
        int[] nums = { 1, 2, 3, 1 };
        int k = 3;
        System.out.println(containsNearbyDuplicate(nums, k));
    }
}
